src/Weather.cpp

```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include "Weather.h"
+
+//constructor & destructor
+Weather::Weather() {}
+Weather::Weather(int year,int month,int Date,weekDay day, string city, string country, int temp, double humidity, double windSpeed,type wtype) :
+year(year),month(month),Date(Date),day(day), city(city), country(country), temp(temp), humidity(humidity), wind_speed(windSpeed),wtype(wtype) {
+}
+Weather::~Weather() {}
+
+//getters & setters
+weekDay Weather::getDay(){
+    return day;
+}
+void Weather::setDay(weekDay day) {
+    Weather::day = day;
+}
+type Weather::getWeathertype() {
+    return Weather::wtype;
+}
+string Weather::getCity(){
+    return city;
+}
+void Weather::setCity(string city){
+    Weather::city = city;
+}
+string Weather::getCountry(){
+    return country;
+}
+void Weather::setCountry(string country){
+    Weather::country = country;
+}
+int Weather::getTemp(){
+    return temp;
+}
+void Weather::setTemp(int temp){
+    Weather::temp = temp;
+}
+double Weather::getHumidity(){
+    return humidity;
+}
+void Weather::setHumidity(double humidity){
+    Weather::humidity = humidity;
+}
+double Weather::getWindSpeed(){
+    return wind_speed;
+}
+void Weather::setWindSpeed(double windSpeed){
+    wind_speed = windSpeed;
+}
+string Weather::getTypeName(){//get type of weather in str from Enum type
+    switch (wtype) {
+        case 0: return "SUNNY";
+        case 1: return "CLOUD";
+        case 2: return "RAINY";
+        case 3: return "FOGGY";
+    }
+    return "noun";
+}
+int Weather::getDate() {
+    return Date;
+}
+
+// methods
+void Weather::show_stats(){//print in console
+    string l;
+    int Max = 50;
+    int Y = (l = to_string(year)).length();
+    int M = (l = to_string(month)).length();
+    int D = getDayName().length();
```

Comment:
not clear variables , please change name

Reply:
changed